specs/0121-translation-memory-management.md

```diff
@@ -0,0 +1,43 @@
+- Feature Name: Translation Memory Management
+- Created: 2024-08-28
+- Associated Issue: #3031
+
+# Summary
+
+Add ability to edit, delete and upload Translation Memory (TM) entries.
+
+# Motivation
+
+Translation memory is a fundamental tool for faster and more consistent translations, and has become even more important with pretranslation, since it's used to train custom machine translation engines in Google AutoML Translation.
+
+Support for TM in Pontoon is limited:
+* Localizers cannot edit or delete existing TM entries directly.
+* TM entries can be changed indirectly via translation workspace, but only if the corresponding strings are still available for translation.
+* Localizers cannot upload their own translation memories.
+
+# Feature explanation
+
+A new "Translation Memory" tab is added to the Team pages, which lists all TM entries for the team. For each entry, source string and translation are provided, clicking on which leads to the translation workspace for the corresponding strings (using the list `query` parameter). Entries are listed in alphabetic order and only shown once for each unique combination of source string and translation.
```

Comment:
Should the entries be listed in alphabetic order of source strings (easier to find similar strings) or in the chronological order of translations (easier to fix recent mistakes)?

Reply:
Personal opinion: alphabetical doesn't seem useful considering the number of entries there will be. If I know which string I'm looking for I'd be using the search function instead of scrolling. At least with chronological there might be some utility.

Reply:
+1 on reverse chronological order

Reply:
Actually it will be hard to sort by the chronological order when items are grouped.

Reply:
Dropped the mention of the order.